### PR TITLE
Fix : Appended Bor Tx in eth_getTransactionByBlockNumberAndIndex

### PR DESCRIPTION
**Pull request by 0xsharma**
https://github.com/maticnetwork/bor/issues/369

### CODE REVIEW

File: internal/ethapi/api.go [hunk 1]
```diff
@@ -1431,16 +1432,26 @@ func newRPCPendingTransaction(tx *types.Transaction, current *types.Header, conf
...
+		return nil
 	}
+
+	tx, _, _, _ := rawdb.ReadBorTransaction(db, borReceipt.TxHash)
+
+	if tx == nil {
+		return nil
+	}
+
+	txs = append(txs, tx)
```
Comment (JekaMas) [inline internal/ethapi/api.go:1449]: Is this slice is sorted after the operation? If not, how could we get by index?

Comment (0xsharma) [reply internal/ethapi/api.go:1449]: The index of state-sync-transaction is always the last one in the block. That's why I believe it is safe to append it to the end of the list.

Comment (JekaMas) [reply internal/ethapi/api.go:1449]: Got it. Should work, however could you add a small test to document expected behaviour?

**Comment by 0xsharma** [comment]
Fixed it in PR https://github.com/maticnetwork/bor/pull/480.

**Pull request closed without merging**